src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -96,30 +96,39 @@ function addModeListener() {
 }
 
 /*******************************************************************************
- * TOC interactivity
+ * Right sidebar table of contents (TOC) interactivity
  */
+function setupPageTableOfContents() {
+  const pageToc = document.querySelector("#pst-page-toc-nav");
+  pageToc.addEventListener("click", (event) => {
+    if (!event.target.matches(".nav-link")) {
+      return;
+    }
+    const clickedLink = event.target;
```

Comment:
This works as long as the user clicks _specifically_ on the `<a>` element. In some cases there can also be e.g. `<code><pre>` as a child element. Clicking on the child element doesn't update the styles as intended because the target isn't `.nav-link`. I think if you use `closest` it should handle it correctly, though:

```suggestion
    const clickedLink = event.target.closest(".nav-link")
    if (!clickedLink) {
      return;
    }
```

Reply:
Good catch!